Katello/katello
# Fixes #34235 - reindex repository after recreating it

#### What are the changes introduced in this pull request?
When a user runs the correct_repositories rake task, and a library repo is found to be missing, it is re-created.  However! if content is indexed into it, it will remain indexed even though the repository is empty.  By reindexing it after creation, we can reset the indexed content.

#### Considerations taken when implementing this change?

#### What are the testing steps for this pull request?
1) create a yum repository and sync it (i used https://jlsherrill.fedorapeople.org/fake-repos/needed-errata/) , notice the content counts show errata & rpms 
2) in the rails console, find the repo href:  `Katello::Repository.find(9).root.repository_references.first.repository_href`
3) Delete the repo:  
```
curl -X DELETE https://`hostname`/pulp/api/v3/repositories/rpm/rpm/920e7484-212c-4009-8324-3f77fe46caab/ --cert /etc/pki/katello/certs/pulp-client.crt --key /etc/pki/katello/private/pulp-client.key
```
4) run:   bundle exec rake katello:correct_repositories COMMIT=true
5) without this pr, the content counts (rpm, errata) stay the same, with it, they are reset to zero
